### DISCUSSION TITLE
Elastic or other transition curves

### DISCUSSION THREAD

**Discussion by appinteractive** [fujidaiti/smooth_sheets#46] [Ideas]
How can I implement an elastic effect when opening the sheet be for the snapping? Discord does have a nice implemention of such effect. 

Currently there is only one curve for opening and closing so I could not just use the elastic curve as the closing effect would be broken. 

And for the physics there is no it do I miss something?

**Comment by fujidaiti** [comment]
>Discord does have a nice implemention of such effect.

Could you post a screen capture of this?

**Reply by nelly833** [reply]
experience that Verna went through didn’t bring her down, but rather she saw this as a reference point to give back and to help people who have been down a similar path. [Verna Haywood](https://verna-haywood.com/)

**Reply by alexa9099** [reply]
Some traders prefer offshore because they offer higher leverage and fewer restrictions. But are they really safe? Have you faced any withdrawal or regulation issues with offshore brokers? Share your stories and advice!

https://www.earnforex.com/ar

**Comment by appinteractive** [comment]
Sure, here you can see the slight overshoot behavior. It's nice if your app needs some more character.

https://github.com/fujidaiti/smooth_sheets/assets/432828/d55f858d-b9a8-4ff0-9f0e-f08ca8d3738a

**Reply by fujidaiti** [reply]
Thanks!

>And for the physics there is no it do I miss something?

No, unfortunately. I'll add this to the roadmap (#51).

**Reply by appinteractive** [reply]
I think that a curve for the snapping physics is also needed or is the given "open" curve used then when you release the sheet for snapping?

**Reply by fujidaiti** [reply]
Ah sorry, I misunderstood what you said. 
Currently, the physics uses [ScrollSpringSimulation](https://api.flutter.dev/flutter/physics/ScrollSpringSimulation-class.html) for the snapping animation by default, which a curve cannot be applied to. However, you can easily create a custom physics by extending `SheetPhysics` or one of its subclasses. Here's an example of a custom physics that has the snapping behavior and uses a curve-applied animation when it snaps:

```dart
import 'package:flutter/physics.dart';
import 'package:smooth_sheets/smooth_sheets.dart';

class CustomSheetPhysics extends StretchingSheetPhysics {
  const CustomSheetPhysics({
    super.parent,
    super.frictionCurve,
    super.stretchingRange,
    this.snappingCurve = Curves.easeInOut,
     // If you want to have multiple snap positions, use SnapToNearest instead.
    this.snappingBehavior = const SnapToNearestEdge(),
  });

  final Curve snappingCurve;
  final SnappingSheetBehavior snappingBehavior;

  @override
  Simulation? createBallisticSimulation(double velocity, SheetMetrics metrics) {
    final snapPixels = snappingBehavior.findSnapPixels(velocity, metrics);
    if (snapPixels != null &&
        !nearEqual(
          snapPixels,
          metrics.pixels,
          Tolerance.defaultTolerance.distance,
        )) {
      return InterpolationSimulation(
        start: metrics.pixels,
        end: snapPixels,
        curve: snappingCurve,
        durationInSeconds: 0.3, // 300ms
      );
    } else {
      return super.createBallisticSimulation(velocity, metrics);
    }
  }
}
```

The `InterpolationSimulation` will be available from v0.3.3.

**Comment by audreyAva12** [comment]
Thank you for considering our options! We highly recommend the versatility and durability of [Jacquard Elastic Band](https://austintrim.co/jacquard-elastic-band/). Their unique design and stretchability make them perfect for various applications. We believe they would be an excellent choice for your needs. If you have any further questions or need assistance, feel free to reach out. We're here to help!

**Comment by mickal22** [comment]
Kingkind is a premier CBD store in San Diego, offering a curated selection of high-quality CBD products. Their focus on purity and potency ensures customers receive effective, reliable relief. [KingKind ](https://www.kingkind.co/)Explore their range for a trusted approach to wellness and relaxation.

**Comment by nelly833** [comment]
experience that Verna went through didn’t bring her down, but rather she saw this as a reference point to give back and to help people who have been down a similar path. [Verna Haywood](https://verna-haywood.com/)

**Comment by mickal22** [comment]
Cannabidiol (CBD) is a non-psychoactive compound found in cannabis that offers potential therapeutic benefits, such as reducing anxiety and pain. It's commonly used in oils, edibles, and creams for wellness purposes. https://www.kingkind.co/
